aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -802,15 +842,115 @@ internal class RealAWSCognitoAuthPlugin(
+                        }
+                    }
+                }
+                else -> onError.accept(AuthException.InvalidStateException())
+            }
+        }
+    }
+
+    private suspend fun updateUserAttributes(
+        attributes: MutableList<AuthUserAttribute>,
+        userAttributesOptions: AWSCognitoAuthUpdateUserAttributesOptions?
+    ):
+        MutableMap<AuthUserAttributeKey, AuthUpdateAttributeResult> {
+            val accessToken = getSession().userPoolTokens.value?.accessToken
+            var userAttributes = attributes.map {
+                AttributeType.invoke {
+                    name = it.key.keyString
+                    value = it.value
+                }
+            }
+            val userAttributesRequest = UpdateUserAttributesRequest.invoke {
+                this.accessToken = accessToken
+                this.userAttributes = userAttributes
+                this.clientMetadata = userAttributesOptions?.metadata
+            }
+            val userAttributeResponse = authEnvironment.cognitoAuthService
+                .cognitoIdentityProviderClient?.updateUserAttributes(
+                    userAttributesRequest
+                )
+            return getUpdateUserAttributeResult(userAttributeResponse, userAttributes)
+        }
+
+    private fun getUpdateUserAttributeResult(
+        response: UpdateUserAttributesResponse?,
+        userAttributeList: List<AttributeType>
+    ): MutableMap<AuthUserAttributeKey, AuthUpdateAttributeResult> {
+
+        val finalResult = HashMap<AuthUserAttributeKey, AuthUpdateAttributeResult>()
+
+        response?.codeDeliveryDetailsList?.let {
+            val codeDeliveryDetailsList = it
+            for (item in codeDeliveryDetailsList) {
+                item.attributeName?.let {
+                    var deliveryMedium = when (item.deliveryMedium) {
+                        DeliveryMediumType.Email -> AuthCodeDeliveryDetails.DeliveryMedium.EMAIL
+                        DeliveryMediumType.Sms -> AuthCodeDeliveryDetails.DeliveryMedium.SMS
+                        else -> AuthCodeDeliveryDetails.DeliveryMedium.UNKNOWN
+                    }
+                    val authCodeDeliveryDetails = AuthCodeDeliveryDetails(
+                        item.destination.toString(),
+                        deliveryMedium,
+                        item.attributeName
+                    )
+                    val nextStep = AuthNextUpdateAttributeStep(
+                        AuthUpdateAttributeStep.CONFIRM_ATTRIBUTE_WITH_CODE,
+                        HashMap(),
+                        authCodeDeliveryDetails
+                    )
+                    val updateAttributeResult = AuthUpdateAttributeResult(false, nextStep)
+                    finalResult[AuthUserAttributeKey.custom(item.attributeName)] = updateAttributeResult
+                }
+            }
+        }
+
+        // Check if all items are added to the dictionary
+        for (item in userAttributeList) {
```

Comment:
Does this mean attributes that were successfully updated are not part of the response and are omitted instead by Cognito?

Reply:
Yes correct